app/models/host/managed.rb

```diff
@@ -231,6 +239,12 @@ def self.model_name
     ActiveModel::Name.new(Host)
   end
 
+  def self.smart_proxy_ids(hosts_scope = Host::Managed.where(nil))
+    hosts_scope.eager_load(smart_proxy_references).map do |host|
+      smart_proxy_references.map { |ref| host.send(ref).try(:id) }.flatten.uniq.compact
```

Comment:
The `.flatten.uniq.compact` should be after the `end` in the next line I think.
If I read the code correctly, line 244 produces an array of proxy ids for a single host `[2,3,nil, nil,3]` and the `map` on line 243 aggregates it to a single two dimensional array: `[[2,3,nil, nil,3], [1,2,nil, 1,1], [nil,nil,nil, nil,5]]`. This array should be flattened, uniqued and compacted.